server/filestore.go

```diff
@@ -3744,18 +3850,31 @@ func (fs *fileStore) storeRawMsg(subj string, hdr, msg []byte, seq uint64, ts in
 	fs.enforceMsgLimit()
 	fs.enforceBytesLimit()
 
+	// Per-message TTL.
+	if fs.ttls != nil && ttl > 0 {
```

Comment:
Should we create fs.ttls here if nil?

Reply:
`fs.ttls` is only non-nil when the `AllowMsgTTL` stream option is enabled.

Reply:
Famous last words ;)